<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GenomicDistributions demo • GenomicDistributionsBioC2022</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GenomicDistributions demo">
<meta property="og:description" content="GenomicDistributionsBioC2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicDistributionsBioC2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomicDistributions_vignette.html">GenomicDistributions demo</a>
    </li>
    <li>
      <a href="../articles/HOWTO_BUILD_WORKSHOP.html">How To Build A Workshop Package</a>
    </li>
    <li>
      <a href="../articles/workshop_example.html">An Example Workshop</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kkupkova/GenomicDistributions_BioC2022" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GenomicDistributions demo</h1>
                        <h4 data-toc-skip class="author">Kristyna
Kupkova<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/databio/GenomicDistributions/blob/HEAD/vignettes/GenomicDistributions_vignette.Rmd" class="external-link"><code>vignettes/GenomicDistributions_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicDistributions_vignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="fast-easy-and-flexible-summary-and-visualization-of-genomic-regions">Fast, Easy, And Flexible Summary And Visualization Of Genomic
Regions<a class="anchor" aria-label="anchor" href="#fast-easy-and-flexible-summary-and-visualization-of-genomic-regions"></a>
</h2>
<p>Authors: Kristyna Kupkova<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, Jose Verdezoto Mosquera<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, Jason P. Smith<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>, Michał
Stolarczyk<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>, Tessa L. Danehy<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>, John T. Lawson<a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a>, Bingjie
Xue<a href="#fn8" class="footnote-ref" id="fnref8"><sup>8</sup></a>,
John T. Stubbs, IV<a href="#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a>, Nathan LeRoy<a href="#fn10" class="footnote-ref" id="fnref10"><sup>10</sup></a>, Nathan C.
Sheffield<a href="#fn11" class="footnote-ref" id="fnref11"><sup>11</sup></a>. <br> Last modified: 19 July, 2022.</p>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<div class="section level4">
<h4 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h4>
<p>During this workshop participants will be introduced to
<em>GenomicDistributions</em> package. At first, we will go over a
general introduction to <em>GenomicDistributions</em>, where we will
learn about the purpose, design, and advantages of
<em>GenomicDistributions</em>. Followed by instructor-led demo, we will
go through all of the functions included in
<em>GenomicDistributions</em>, where participants will be able to try
the functions on provided example datasets. The participants will learn,
how to calculate and plot summary statistics extracted from the example
datasets and how to further customize plots generated with
<em>GenomicDistributions</em>. Some of the <em>GenomicDistributions</em>
functions require genome annotations (for <em>hg19</em>, <em>hg38</em>,
and <em>mm10</em> these are all included in the
<em>GenomicDistributionsData</em> package and passed to a subset of
functions ending with the suffix <em>“Ref”</em>), the participants will
be taught how to easily create and use these annotations. At the end of
the workshop, all participants will be able to extract following summary
statistics about their regions of interest: distribution over
chromosomes, distances from genomic features (e.g., TSSs), distribution
over annotation classes, signal summary in regions, region widths,
distances to the nearest neighbor, GC content, and dinucleotide
frequencies.</p>
</div>
<div class="section level4">
<h4 id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h4>
<p>Helpful but not required:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the GenomicRanges class</li>
</ul>
<p>Anyone interested in getting familiar with
<em>GenomicDistributions</em> before the workshop is welcome to that
that by reading the <em>GenomicDistributions</em> publication or
vignettes provided with the package:</p>
<ul>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-022-08467-y" class="external-link">publication</a></li>
<li><a href="http://code.databio.org/GenomicDistributions/articles/index.html" class="external-link">vignettes</a></li>
</ul>
</div>
<div class="section level4">
<h4 id="participation">Participation<a class="anchor" aria-label="anchor" href="#participation"></a>
</h4>
<p>Participants will be able to test all of the functions provided in
<em>GenomicDistributions</em> on example datasets, learn how to
customize plots and create genome annotations required by some of the
functions.</p>
</div>
<div class="section level4">
<h4 id="r-bioconductor-packages-used">
<em>R</em> / <em>Bioconductor</em> packages used<a class="anchor" aria-label="anchor" href="#r-bioconductor-packages-used"></a>
</h4>
<ul>
<li><em>GenomicDistributions</em></li>
<li><em>GenomicDistributionsData</em></li>
</ul>
</div>
<div class="section level4">
<h4 id="time-outline">Time outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h4>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Package demo</td>
<td>25m</td>
</tr>
<tr class="odd">
<td>Questions</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h4>
<ul>
<li>calculate and plot summary statistics about genomic regions</li>
<li>create genome annotations from FASTA and GTF files</li>
</ul>
</div>
<div class="section level4">
<h4 id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<p>During the workshop the participants will learn how to get following
information about regions of interest:</p>
<ul>
<li>distribution over chromosomes</li>
<li>distances from genomic features (e.g., TSSs)</li>
<li>distribution over annotation classes</li>
<li>signal summary in regions</li>
<li>region widths</li>
<li>distances to the nearest neighbor</li>
<li>GC content</li>
<li>dinucleotide frequencies</li>
</ul>
<p>Participants will also learn how to:</p>
<ul>
<li>create custom annotation files</li>
<li>customize plots</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="workshop">Workshop<a class="anchor" aria-label="anchor" href="#workshop"></a>
</h3>
</div>
<div class="section level3">
<h3 id="a--load-data">A. Load data<a class="anchor" aria-label="anchor" href="#a--load-data"></a>
</h3>
<p>Let’s first load our example region sets (BED files - hg38 genome
assembly) as <em>GRanges</em> / <em>GRangesList</em> objects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://kkupkova.github.io/GenomicDistributionsBioC2022/" class="external-link">"GenomicDistributionsBioC2022"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># provide full path to genomic regions of interest (BED files)</span></span>
<span><span class="va">H3K27ac_Bcell_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg38_H3K27ac_Bcell.bed.gz"</span>, </span>
<span>                                 package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span><span class="va">H3K27me3_Bcell_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg38_H3K27me3_Bcell.bed.gz"</span>, </span>
<span>                                  package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span><span class="va">FGF2_iPSC_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg38_FGF2_iPSC.bed.gz"</span>, </span>
<span>                             package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># import bed files as GenomicRange objects</span></span>
<span><span class="va">H3K27ac_Bcell</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">H3K27ac_Bcell_file</span><span class="op">)</span></span>
<span><span class="va">H3K27me3_Bcell</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">H3K27me3_Bcell_file</span><span class="op">)</span></span>
<span><span class="va">FGF2_iPSC</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">FGF2_iPSC_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove unknown chromosomes:</span></span>
<span><span class="va">H3K27ac_Bcell</span> <span class="op">=</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">H3K27ac_Bcell</span>,pruning.mode<span class="op">=</span><span class="st">"coarse"</span><span class="op">)</span></span>
<span><span class="va">H3K27me3_Bcell</span> <span class="op">=</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">H3K27me3_Bcell</span>,pruning.mode<span class="op">=</span><span class="st">"coarse"</span><span class="op">)</span></span>
<span><span class="va">FGF2_iPSC</span> <span class="op">=</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">FGF2_iPSC</span>,pruning.mode<span class="op">=</span><span class="st">"coarse"</span><span class="op">)</span></span></code></pre></div>
<p>We now have our individual datasets in form of a <em>GRanges</em>
objects. Let’s have a look.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">FGF2_iPSC</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames        ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     chr1 629777-630122      *</span></span>
<span><span class="co">#&gt;   [2]     chr1 633847-634215      *</span></span>
<span><span class="co">#&gt;   [3]     chr1 778575-778853      *</span></span>
<span><span class="co">#&gt;   [4]     chr1 827332-827648      *</span></span>
<span><span class="co">#&gt;   [5]     chr1 921142-921392      *</span></span>
<span><span class="co">#&gt;   [6]     chr1 940207-940475      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>To enable comparison of multiple region sets we need to combine the
individual <em>GRanges</em> objects into a <em>GRangesList</em>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GRList</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></span><span class="op">(</span>H3K27ac_Bcell<span class="op">=</span><span class="va">H3K27ac_Bcell</span>,</span>
<span>                     H3K27me3_Bcell<span class="op">=</span><span class="va">H3K27me3_Bcell</span>,</span>
<span>                     FGF2_iPSC<span class="op">=</span><span class="va">FGF2_iPSC</span><span class="op">)</span></span>
<span></span>
<span><span class="va">GRList</span></span>
<span><span class="co">#&gt; GRangesList object of length 3:</span></span>
<span><span class="co">#&gt; $H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; GRanges object with 58033 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames              ranges strand</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;       [1]     chr1     1000310-1000753      *</span></span>
<span><span class="co">#&gt;       [2]     chr1 100037011-100040452      *</span></span>
<span><span class="co">#&gt;       [3]     chr1 100040882-100041361      *</span></span>
<span><span class="co">#&gt;       [4]     chr1 100046295-100047961      *</span></span>
<span><span class="co">#&gt;       [5]     chr1 100048276-100048828      *</span></span>
<span><span class="co">#&gt;       ...      ...                 ...    ...</span></span>
<span><span class="co">#&gt;   [58029]     chrX   97563113-97564372      *</span></span>
<span><span class="co">#&gt;   [58030]     chrX   97564422-97564749      *</span></span>
<span><span class="co">#&gt;   [58031]     chrX       989528-991924      *</span></span>
<span><span class="co">#&gt;   [58032]     chrX     9953085-9953661      *</span></span>
<span><span class="co">#&gt;   [58033]     chrX     9980613-9982237      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $H3K27me3_Bcell</span></span>
<span><span class="co">#&gt; GRanges object with 51085 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames              ranges strand</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;       [1]     chr1     1000432-1000779      *</span></span>
<span><span class="co">#&gt;       [2]     chr1     1000882-1001645      *</span></span>
<span><span class="co">#&gt;       [3]     chr1     1002104-1002649      *</span></span>
<span><span class="co">#&gt;       [4]     chr1     1002885-1005766      *</span></span>
<span><span class="co">#&gt;       [5]     chr1 100336992-100337457      *</span></span>
<span><span class="co">#&gt;       ...      ...                 ...    ...</span></span>
<span><span class="co">#&gt;   [51081]     chrY     2830500-2830755      *</span></span>
<span><span class="co">#&gt;   [51082]     chrY     2834328-2834614      *</span></span>
<span><span class="co">#&gt;   [51083]     chrY     2899471-2899699      *</span></span>
<span><span class="co">#&gt;   [51084]     chrY     7712260-7712492      *</span></span>
<span><span class="co">#&gt;   [51085]     chrY     8417617-8417831      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FGF2_iPSC</span></span>
<span><span class="co">#&gt; GRanges object with 9974 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames            ranges strand</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;      [1]     chr1     629777-630122      *</span></span>
<span><span class="co">#&gt;      [2]     chr1     633847-634215      *</span></span>
<span><span class="co">#&gt;      [3]     chr1     778575-778853      *</span></span>
<span><span class="co">#&gt;      [4]     chr1     827332-827648      *</span></span>
<span><span class="co">#&gt;      [5]     chr1     921142-921392      *</span></span>
<span><span class="co">#&gt;      ...      ...               ...    ...</span></span>
<span><span class="co">#&gt;   [9970]     chrY 10999752-11000089      *</span></span>
<span><span class="co">#&gt;   [9971]     chrY 11305156-11305354      *</span></span>
<span><span class="co">#&gt;   [9972]     chrY 11329757-11329976      *</span></span>
<span><span class="co">#&gt;   [9973]     chrY 11721851-11722119      *</span></span>
<span><span class="co">#&gt;   [9974]     chrY 11731154-11731384      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Now that we have our example data loaded, let’s see how can we
extract information from these files using
<em>GenomicDistributions</em>.</p>
</div>
<div class="section level3">
<h3 id="b--distribution-over-chromosomes">B. Distribution over chromosomes<a class="anchor" aria-label="anchor" href="#b--distribution-over-chromosomes"></a>
</h3>
<div class="section level5">
<h5 id="calcchrombinsref-plotchrombins">
<em>calcChromBinsRef()</em> + <em>plotChromBins()</em><a class="anchor" aria-label="anchor" href="#calcchrombinsref-plotchrombins"></a>
</h5>
<p>Let’s start by calculating and plotting the distribution of regions
over chromosomes for a single genomic region set (single
<em>GRanges</em> object).</p>
<p><strong>Calculate the distribution over chromosomes.</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDistribution_FGF2</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcChromBinsRef.html" class="external-link">calcChromBinsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">FGF2_iPSC</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">chromDistribution_FGF2</span><span class="op">)</span></span>
<span><span class="co">#&gt;     chr   start     end regionID withinGroupID  N</span></span>
<span><span class="co">#&gt; 1: chr1       1 1068482        1             1 11</span></span>
<span><span class="co">#&gt; 2: chr1 1068483 2136965        2             2 18</span></span>
<span><span class="co">#&gt; 3: chr1 2136966 3205447        3             3 15</span></span>
<span><span class="co">#&gt; 4: chr1 3205448 4273930        4             4  7</span></span>
<span><span class="co">#&gt; 5: chr1 5342413 6410895        6             6  3</span></span>
<span><span class="co">#&gt; 6: chr1 6410896 7479377        7             7  9</span></span></code></pre></div>
<p>The output from the <em>calcChromBinsRef</em> (or
<em>calcChromBins</em> that we will cover later in the workshop) is a
data.table with following columns: <em>chr</em>, <em>start</em>,
<em>end</em> - coordinates of a bin, <em>regionID</em> - bin number in a
sorted list ob bins, <em>withinGroupID</em> - bin number within a given
chromosome, <em>N</em> - number of region falling within the bin.</p>
<p><strong>Plot results.</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotChromBins.html" class="external-link">plotChromBins</a></span><span class="op">(</span><span class="va">chromDistribution_FGF2</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-plots-single-1.png" width="480"></p>
<p><strong>Process multiple region sets</strong></p>
<p><em>GenomicDistributions</em> functions are able to process multiple
region sets at once. Let’s try that by passing <em>GRangesList</em>
object to the same function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDistributionMulti</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcChromBinsRef.html" class="external-link">calcChromBinsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">chromDistributionMulti</span><span class="op">)</span></span>
<span><span class="co">#&gt;     chr   start     end regionID withinGroupID  N          name</span></span>
<span><span class="co">#&gt; 1: chr1       1 1068482        1             1  8 H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 2: chr1 1068483 2136965        2             2 95 H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 3: chr1 2136966 3205447        3             3 46 H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 4: chr1 3205448 4273930        4             4 20 H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 5: chr1 4273931 5342412        5             5  1 H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 6: chr1 5342413 6410895        6             6  6 H3K27ac_Bcell</span></span></code></pre></div>
<p>As you can see, the output of the <em>calc</em> function has now an
additional <em>name</em> column that assigns individual values to a
dataset of origin. But plotting function stays the same, let’s try
it.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotChromBins.html" class="external-link">plotChromBins</a></span><span class="op">(</span><span class="va">chromDistributionMulti</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-plots-multi-1.png" width="480"></p>
<p>All of the functions in <em>GenomicDistributions</em> have the same
design – you can process single genomic region set as <em>GRanges</em>
object, or process multiple region sets with the same functions by
passing them <em>GRangesList</em>. All following examples will be
therefore shown only on the created <em>GRangesList</em> for multiple
comparisons.</p>
</div>
</div>
<div class="section level3">
<h3 id="c--distances-from-tsss">C. Distances from TSSs<a class="anchor" aria-label="anchor" href="#c--distances-from-tsss"></a>
</h3>
<div class="section level5">
<h5 id="calcfeaturedistreftss-plotfeaturedist">
<em>calcFeatureDistRefTSS()</em> + <em>plotFeatureDist()</em><a class="anchor" aria-label="anchor" href="#calcfeaturedistreftss-plotfeaturedist"></a>
</h5>
<p>To calculate distances from genomic features,
<em>GenomicDistributions</em> offers <em>calcFeature</em> function
(example shown in custom features section). Distance from transcription
start sites (TSSs) is commonly used for analysis of genomic regions, we
therefore created <em>calcFeatureDistRefTSS</em>, which provides
precompiled set of TSSs for <em>hg19</em>, <em>hg38</em>, <em>mm9</em>,
and <em>mm10</em>. Let’s give this function a try.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TSSdist</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcFeatureDistRefTSS.html" class="external-link">calcFeatureDistRefTSS</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotFeatureDist.html" class="external-link">plotFeatureDist</a></span><span class="op">(</span><span class="va">TSSdist</span>, featureName <span class="op">=</span> <span class="st">"TSS"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-TSS-multi-1.png" width="480"></p>
<p>The _ plotFeatureDist_ plotting function (like many other
<em>GenomicDistributions</em> plotting functions) offers multiple
plotting options. In default setting plotted are regions within +- 100
kb from TSSs. Let’s try what happens when we change few of the
parameters in the _ plotFeatureDist_ function.</p>
<p><strong><em>Exercise 1</em></strong></p>
<p>Using the previously calculated _ TSSdist_ object, plot histogram of
distances to TSSs within 5 kb from TSSs (argument <em>size</em>), where
regions further than +-5 kb will be accumulated and plotted into
“infinite bins” (argument <em>infBins</em>).</p>
</div>
</div>
<div class="section level3">
<h3 id="d--distribution-over-genomic-partitions">D. Distribution over genomic partitions<a class="anchor" aria-label="anchor" href="#d--distribution-over-genomic-partitions"></a>
</h3>
<div class="section level5">
<h5 id="calcpartitionsref-plotpartitions">
<em>calcPartitionsRef()</em> + <em>plotPartitions()</em><a class="anchor" aria-label="anchor" href="#calcpartitionsref-plotpartitions"></a>
</h5>
<p>To calculate and plot, how are genomic regions distributed across
genomic partitions such as promoters, exons, introns, 5’UTR, and 3’UTR
<em>GenomicDistributions</em> provides <em>calcPartitionsRef</em> and
<em>plotPartitions</em> functions. <em>calcPartitions</em> function
requires provided annotation classes (partitions), which are again
provided within <em>calcPartitionsRef</em> function and we will show how
to generate these later in the custom reference section.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">partitions</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcPartitionsRef.html" class="external-link">calcPartitionsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotPartitions.html" class="external-link">plotPartitions</a></span><span class="op">(</span><span class="va">partitions</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/partitions-1.png" width="480"></p>
</div>
<div class="section level5">
<h5 id="calcexpectedpartitionsref-plotexpectedpartitions">
<em>calcExpectedPartitionsRef()</em> +
<em>plotExpectedPartitions()</em><a class="anchor" aria-label="anchor" href="#calcexpectedpartitionsref-plotexpectedpartitions"></a>
</h5>
<p>Partition plots do not account for different sizes of individual
partitions. If you want to account for the different sizes of individual
partitions, we recommend that you use
<em>calcExpectedPartitionsRef()</em> and
<em>plotExpectedPartitions()</em> functions to calculate and plot if
observed overlap is higher or lower to what we would expect based on
size contribution of individual partitions to genome size. Let’s have a
look!</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expPartitions</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcExpectedPartitionsRef.html" class="external-link">calcExpectedPartitionsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotExpectedPartitions.html" class="external-link">plotExpectedPartitions</a></span><span class="op">(</span><span class="va">expPartitions</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/expPartitions-1.png" width="480"></p>
</div>
<div class="section level5">
<h5 id="calccumulativepartitionsref-plotcumulativepartitions">
<em>calcCumulativePartitionsRef()</em> +
<em>plotCumulativePartitions()</em><a class="anchor" aria-label="anchor" href="#calccumulativepartitionsref-plotcumulativepartitions"></a>
</h5>
<p>To further show, how are genomic regions accumulating in different
partitions (genomic regions are sorted from smallest to largest),
<em>GenomicDistributions</em> provide a unique set of functions to do
that: _ calcCumulativePartitionsRef_ and
<em>plotCumulativePartitions</em>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cumPartitions</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcCumulativePartitionsRef.html" class="external-link">calcCumulativePartitionsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotCumulativePartitions.html" class="external-link">plotCumulativePartitions</a></span><span class="op">(</span><span class="va">cumPartitions</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/cumPartitions-1.png" width="768"></p>
</div>
</div>
<div class="section level3">
<h3 id="e--signal-summary-in-regions">E. Signal summary in regions<a class="anchor" aria-label="anchor" href="#e--signal-summary-in-regions"></a>
</h3>
<div class="section level5">
<h5 id="calcsummarysignal-plotsummarysignal">
<em>calcSummarySignal()</em> + <em>plotSummarySignal()</em><a class="anchor" aria-label="anchor" href="#calcsummarysignal-plotsummarysignal"></a>
</h5>
<p>In this section, we will learn how to calculate summary of signal
values within our regions sets. <em>GenomicDistributions</em> do not
provide <em>”Ref”</em> functions to calculate the signal in the
overlaps. The <em>calcSummarySignal</em> function requires a signal
matrix, where each column is a condition of interest, each row a defined
region within a genome and values are a normalized signal. The first
column in the matrix contains the coordinates for a given region.
<em>GenomicDistributionsData</em> package, provides an example of such
matrix (<em>hg19</em>, <em>hg38</em>, <em>mm10</em>) with calculated
normalized chromatin accessibility signal across different cell types.
Let’s have a look at the matrix for <em>hg38</em>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load matrix with chromatin accessibility values for different cell types from </span></span>
<span><span class="co"># GenomicDistributionsData package</span></span>
<span><span class="va">openSignalMatrix_hg38</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicDistributionsData/man/openSignalMatrix_hg38.html" class="external-link">openSignalMatrix_hg38</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># and let's have a look at few top rows and columns</span></span>
<span><span class="va">openSignalMatrix_hg38</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                          V1 amniotic_epithelial_cell amniotic_stem_cell</span></span>
<span><span class="co">#&gt; 1: chr1_100002175_100002325                   0.0212             0.0433</span></span>
<span><span class="co">#&gt; 2: chr1_100004335_100004598                   0.0315             0.0304</span></span>
<span><span class="co">#&gt; 3:     chr1_1000055_1000285                   1.0000             1.0000</span></span>
<span><span class="co">#&gt; 4: chr1_100005564_100005714                   0.0202             0.0283</span></span>
<span><span class="co">#&gt; 5: chr1_100005741_100006041                   0.0263             0.0087</span></span>
<span><span class="co">#&gt;    astrocyte astrocyte_of_the_cerebellum</span></span>
<span><span class="co">#&gt; 1:    0.0103                      0.0173</span></span>
<span><span class="co">#&gt; 2:    0.0321                      0.0290</span></span>
<span><span class="co">#&gt; 3:    1.0000                      1.0000</span></span>
<span><span class="co">#&gt; 4:    0.0125                      0.0173</span></span>
<span><span class="co">#&gt; 5:    0.0309                      0.0203</span></span></code></pre></div>
<p>We can now pass this matrix to the function that extracts signal in
the overlapping regions.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># extract values from overlapping regions</span></span>
<span><span class="va">cellSpecTable</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcSummarySignal.html" class="external-link">calcSummarySignal</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, signalMatrix<span class="op">=</span><span class="va">openSignalMatrix_hg38</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cellSpecTable</span><span class="op">$</span><span class="va">signalSummaryMatrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                   queryPeak amniotic_epithelial_cell amniotic_stem_cell</span></span>
<span><span class="co">#&gt; 1:     chr1_1000310_1000753                   0.1622             0.4556</span></span>
<span><span class="co">#&gt; 2: chr1_100037011_100040452                   0.2499             0.3590</span></span>
<span><span class="co">#&gt; 3: chr1_100046295_100047961                   0.0870             0.0494</span></span>
<span><span class="co">#&gt; 4: chr1_100048276_100048828                   0.0523             0.0420</span></span>
<span><span class="co">#&gt; 5: chr1_100049149_100049730                   0.0523             0.0420</span></span>
<span><span class="co">#&gt;    astrocyte astrocyte_of_the_cerebellum</span></span>
<span><span class="co">#&gt; 1:    0.2605                      0.2650</span></span>
<span><span class="co">#&gt; 2:    0.2867                      0.2884</span></span>
<span><span class="co">#&gt; 3:    0.0302                      0.0236</span></span>
<span><span class="co">#&gt; 4:    0.0288                      0.0416</span></span>
<span><span class="co">#&gt; 5:    0.0288                      0.0416</span></span>
<span><span class="va">cellSpecTable</span><span class="op">$</span><span class="va">matrixStats</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;    amniotic_epithelial_cell amniotic_stem_cell astrocyte</span></span>
<span><span class="co">#&gt; 1:                  0.00100            0.00090    0.0011</span></span>
<span><span class="co">#&gt; 2:                  0.04160            0.03790    0.0388</span></span>
<span><span class="co">#&gt; 3:                  0.08290            0.07520    0.0782</span></span>
<span><span class="co">#&gt; 4:                  0.35025            0.42585    0.3562</span></span>
<span><span class="co">#&gt; 5:                  0.81320            1.00000    0.8322</span></span>
<span><span class="co">#&gt;    astrocyte_of_the_cerebellum astrocyte_of_the_hippocampus</span></span>
<span><span class="co">#&gt; 1:                     0.00310                       0.0010</span></span>
<span><span class="co">#&gt; 2:                     0.03840                       0.0398</span></span>
<span><span class="co">#&gt; 3:                     0.07760                       0.0780</span></span>
<span><span class="co">#&gt; 4:                     0.38135                       0.5067</span></span>
<span><span class="co">#&gt; 5:                     0.89520                       1.0000</span></span></code></pre></div>
<p>The output is a list that contains- 1) a matrix with signal values
within our region sets:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellSpecTable</span><span class="op">$</span><span class="va">signalSummaryMatrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">71</span><span class="op">:</span><span class="fl">75</span><span class="op">]</span></span>
<span><span class="co">#&gt;    T-helper_1_cell T-helper_17_cell T-helper_2_cell trophoblast_cell</span></span>
<span><span class="co">#&gt; 1:          0.1579           0.0153          0.1627           0.2565</span></span>
<span><span class="co">#&gt; 2:          1.0000           0.6351          1.0000           0.4297</span></span>
<span><span class="co">#&gt; 3:          0.6212           1.0000          0.4485           0.0949</span></span>
<span><span class="co">#&gt; 4:          0.0968           0.0226          0.0885           0.0407</span></span>
<span><span class="co">#&gt; 5:          0.0968           0.0226          0.0885           0.0407</span></span>
<span><span class="co">#&gt;             name</span></span>
<span><span class="co">#&gt; 1: H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 2: H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 3: H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 4: H3K27ac_Bcell</span></span>
<span><span class="co">#&gt; 5: H3K27ac_Bcell</span></span></code></pre></div>
<p>and 2) a matrix containing boxplot summary statistics for values
within each condition.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cellSpecTable</span><span class="op">$</span><span class="va">matrixStats</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">71</span><span class="op">:</span><span class="fl">75</span><span class="op">]</span></span>
<span><span class="co">#&gt;    T-helper_17_cell T-helper_2_cell trophoblast_cell          name     boxStats</span></span>
<span><span class="co">#&gt; 1:           0.0011          0.0001          0.00080 H3K27ac_Bcell lowerWhisker</span></span>
<span><span class="co">#&gt; 2:           0.0494          0.0611          0.04170 H3K27ac_Bcell   lowerHinge</span></span>
<span><span class="co">#&gt; 3:           0.2017          0.2048          0.08070 H3K27ac_Bcell       median</span></span>
<span><span class="co">#&gt; 4:           0.9920          1.0000          0.38235 H3K27ac_Bcell   upperHinge</span></span>
<span><span class="co">#&gt; 5:           1.0000          1.0000          0.89290 H3K27ac_Bcell upperWhisker</span></span></code></pre></div>
<p>You can pass the output of the <em>calcSummarySignal</em> function
directly to the <em>plotSummarySignal</em> function without any
additional inputs, however, if you want different conditions to be
colored by different colors (e.g., color cell types based on tissue of
origin), you need to provide metadata. Let’s have a look at an example
of suitable metadata matrix. The first column must be names
<em>colName</em> and contain column names from signal matrix (here
different cell types). Notice that the column with region coordinates is
not included in the metadata matrix. The rest of columns are named and
generated by the user. The plotting function then accepts an argument
defining the name of the column used for coloring.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cellType metadata matrix is provided as part of GnomicDistribution data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cellTypeMetadata</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         colName tissueType      group</span></span>
<span><span class="co">#&gt; 1:                          A20      blood   lymphoid</span></span>
<span><span class="co">#&gt; 2:     amniotic_epithelial_cell  embryonic epithelium</span></span>
<span><span class="co">#&gt; 3:           amniotic_stem_cell  embryonic   stemCell</span></span>
<span><span class="co">#&gt; 4:                    astrocyte        CNS  astrocyte</span></span>
<span><span class="co">#&gt; 5:  astrocyte_of_the_cerebellum        CNS  astrocyte</span></span>
<span><span class="co">#&gt; 6: astrocyte_of_the_hippocampus        CNS  astrocyte</span></span></code></pre></div>
<p>Now, let’s finally plot the results!</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotSummarySignal.html" class="external-link">plotSummarySignal</a></span><span class="op">(</span>signalSummaryList<span class="op">=</span><span class="va">cellSpecTable</span>, </span>
<span>                  metadata<span class="op">=</span><span class="va">cellTypeMetadata</span>,</span>
<span>                  colorColumn<span class="op">=</span><span class="st">"tissueType"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/signalSummary-1.png" width="864"></p>
</div>
</div>
<div class="section level3">
<h3 id="f--region-widths">F. Region widths<a class="anchor" aria-label="anchor" href="#f--region-widths"></a>
</h3>
<div class="section level5">
<h5 id="calcwidth-plotqthist">
<em>calcWidth()</em> + <em>plotQTHist()</em><a class="anchor" aria-label="anchor" href="#calcwidth-plotqthist"></a>
</h5>
<p>In this section we will have a look at the widths of regions within
region sets. We do not need any reference annotation data for these
functions. Plotting the distribution of region widths often leads to
long tails due to very large or very small outliers. To overcome this
problem, <em>GenomicDistributions</em> offers <em>plotQTHist</em>
function that accumulates the top and bottom predefined percentiles of
region widths into separate bins. Let’s give it a try.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">widths</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcWidth.html" class="external-link">calcWidth</a></span><span class="op">(</span><span class="va">GRList</span><span class="op">)</span></span>
<span><span class="co"># let's say we want only 10 bins in our histogram and top and bottom 5% </span></span>
<span><span class="co"># of regions will fall into separate bins </span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotQTHist.html" class="external-link">plotQTHist</a></span><span class="op">(</span><span class="va">widths</span>, bins <span class="op">=</span> <span class="fl">10</span>, quantThresh <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/qtHist-1.png" width="768"></p>
</div>
</div>
<div class="section level3">
<h3 id="g--distances-to-the-nearest-neighbor">G. distances to the nearest neighbor<a class="anchor" aria-label="anchor" href="#g--distances-to-the-nearest-neighbor"></a>
</h3>
<div class="section level5">
<h5 id="calcneighbordist-plotneighbordist">
<em>calcNeighborDist()</em> + <em>plotNeighborDist()</em><a class="anchor" aria-label="anchor" href="#calcneighbordist-plotneighbordist"></a>
</h5>
<p>Next functions offered in <em>GenomicDistributions</em> package
calculate and plot the distances between neighboring regions within a
region set. You don’t need any genome annotation here, so let’s just
give it a go.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">neighbors</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcNeighborDist.html" class="external-link">calcNeighborDist</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotNeighborDist.html" class="external-link">plotNeighborDist</a></span><span class="op">(</span><span class="va">neighbors</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/neighbors-1.png" width="384"></p>
</div>
</div>
<div class="section level3">
<h3 id="h--gc-content">H. GC content<a class="anchor" aria-label="anchor" href="#h--gc-content"></a>
</h3>
<div class="section level5">
<h5 id="calcgccontentref-plotgccontent">
<em>calcGCContentRef()</em> + <em>plotGCContent()</em><a class="anchor" aria-label="anchor" href="#calcgccontentref-plotgccontent"></a>
</h5>
<p>In order to calculate the GC content withing your region sets, you
must provide a <em>BSgenome</em> object from a <em>BSGenome</em>
Bioconductor package associated with your organism.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bsg</span> <span class="op">=</span> <span class="va">BSgenome.Hsapiens.UCSC.hg38</span></span>
<span><span class="va">gc</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcGCContent.html" class="external-link">calcGCContent</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, ref <span class="op">=</span> <span class="va">bsg</span><span class="op">)</span> </span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotGCContent.html" class="external-link">plotGCContent</a></span><span class="op">(</span><span class="va">gc</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/GC-1.png" width="384"></p>
</div>
</div>
<div class="section level3">
<h3 id="i--dinucleotide-frequencies">I. Dinucleotide frequencies<a class="anchor" aria-label="anchor" href="#i--dinucleotide-frequencies"></a>
</h3>
<div class="section level5">
<h5 id="calcdinuclfreqref-plotdinuclfreq">
<em>calcDinuclFreqRef()</em> + <em>plotDinuclFreq()</em><a class="anchor" aria-label="anchor" href="#calcdinuclfreqref-plotdinuclfreq"></a>
</h5>
<p>Just like in functions for GC content calculations, in order to
calculate dinucleotide frequencies, you need to provide
<em>BSgenome</em> object.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dinuc</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcDinuclFreq.html" class="external-link">calcDinuclFreq</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, ref<span class="op">=</span><span class="va">bsg</span><span class="op">)</span> </span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotDinuclFreq.html" class="external-link">plotDinuclFreq</a></span><span class="op">(</span><span class="va">dinuc</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/dinuc-1.png" width="576"></p>
</div>
</div>
<div class="section level3">
<h3 id="j--custom-reference">J. Custom reference<a class="anchor" aria-label="anchor" href="#j--custom-reference"></a>
</h3>
<p>Unless you are working with human or mouse data, for which associated
<em>GenomicDistributionsData</em> package provides precompiled
annotation files (available for: <em>hg19</em>, <em>hg38</em>,
<em>mm10</em>, <em>mm9</em>), you will need to use <em>“calc”</em>
function without <em>“Ref”</em> suffix and provide required genome
annotations objects (e.g., chromosome sizes, list of TSSs). But don’t
worry, <em>GenomicDistributions</em> provide functions to create such
annotation objects. All you need to do is provide FASTA and GTF files
for your organism of interest. The only exceptions are functions for
calculation of GC content and dinucleotide frequencies, which require
BSgenome object. We will show, how to pass a BSgenome object to these
functions at the end of this section.</p>
<p>Let’s try to create all necessary genome annotation for
<em>Caenorhabditis elegans</em>.</p>
<p>We can first have a look, what can we extract from FASTA file. All
you need to do is simply provide a path to your FASTA file (or URL) and
our annotation functions will take care of everything else!</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># provide path to the FASTA file</span></span>
<span><span class="va">fastaSource</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ce11.fa.gz"</span>, </span>
<span>                          package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span><span class="va">fastaSource</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmplmfokC/temp_libpath44f4b300e72/GenomicDistributionsBioC2022/extdata/ce11.fa.gz"</span></span></code></pre></div>
<div class="section level5">
<h5 id="get-chromosome-sizes">1. Get chromosome sizes<a class="anchor" aria-label="anchor" href="#get-chromosome-sizes"></a>
</h5>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get chromosome sizes from the FASTA file</span></span>
<span><span class="va">Celegans_chromSizes</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/getChromSizesFromFasta.html" class="external-link">getChromSizesFromFasta</a></span><span class="op">(</span>source<span class="op">=</span><span class="va">fastaSource</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Celegans_chromSizes</span></span>
<span><span class="co">#&gt;        I       II      III       IV        V        X    MtDNA </span></span>
<span><span class="co">#&gt; 15072434 15279421 13783801 17493829 20924180 17718942    13794</span></span></code></pre></div>
<p>As you can see, the output of the <em>getChromSizesFromFasta</em>
function is a named vector. Our FASTA file comes from Ensembl and
therefore misses <em>chr</em> in front of the chromosome number. Let’s
fix that.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># paste chr in front of chromosome numbers</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Celegans_chromSizes</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Celegans_chromSizes</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Celegans_chromSizes</span></span>
<span><span class="co">#&gt;     chrI    chrII   chrIII    chrIV     chrV     chrX chrMtDNA </span></span>
<span><span class="co">#&gt; 15072434 15279421 13783801 17493829 20924180 17718942    13794</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="get-uniformly-sized-bins-across-genome">2. Get uniformly sized bins across genome<a class="anchor" aria-label="anchor" href="#get-uniformly-sized-bins-across-genome"></a>
</h5>
<p>The <em>calcChromBins</em> function requires a <em>GRangesList</em>
object containing uniformly sized bins for each chromosome. Let’s create
these bins using previously generated chromosome sizes.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create GRangesList object with uniformly-sized bins across chromosomes</span></span>
<span><span class="va">Celegans_bins</span>  <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/getGenomeBins.html" class="external-link">getGenomeBins</a></span><span class="op">(</span>chromSizes<span class="op">=</span><span class="va">Celegans_chromSizes</span>, binCount <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">Celegans_bins</span></span>
<span><span class="co">#&gt; GRangesList object of length 7:</span></span>
<span><span class="co">#&gt; $`1`</span></span>
<span><span class="co">#&gt; GRanges object with 150 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames            ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;     1     chrI          1-100483      *</span></span>
<span><span class="co">#&gt;     2     chrI     100484-200966      *</span></span>
<span><span class="co">#&gt;     3     chrI     200967-301449      *</span></span>
<span><span class="co">#&gt;     4     chrI     301450-401932      *</span></span>
<span><span class="co">#&gt;     5     chrI     401933-502414      *</span></span>
<span><span class="co">#&gt;   ...      ...               ...    ...</span></span>
<span><span class="co">#&gt;   146     chrI 14570020-14670501      *</span></span>
<span><span class="co">#&gt;   147     chrI 14670502-14770984      *</span></span>
<span><span class="co">#&gt;   148     chrI 14770985-14871467      *</span></span>
<span><span class="co">#&gt;   149     chrI 14871468-14971950      *</span></span>
<span><span class="co">#&gt;   150     chrI 14971951-15072433      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ...</span></span>
<span><span class="co">#&gt; &lt;6 more elements&gt;</span></span></code></pre></div>
<p>For the next set of genome annotations, we will need a GTF file. The
process is the same as with the FASTA file, just give
<em>GenomicDistributions</em> the path to the GTF file (or URL) and
everything will be taken care of.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># provide path to the GTF file</span></span>
<span><span class="va">gtfSource</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ce11.gtf.gz"</span>, </span>
<span>                          package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span><span class="va">gtfSource</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmplmfokC/temp_libpath44f4b300e72/GenomicDistributionsBioC2022/extdata/ce11.gtf.gz"</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="get-list-of-tsss">3. Get list of TSSs<a class="anchor" aria-label="anchor" href="#get-list-of-tsss"></a>
</h5>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get coordinates of TSSs</span></span>
<span><span class="va">Celegans_TSSs</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/getTssFromGTF.html" class="external-link">getTssFromGTF</a></span><span class="op">(</span>source<span class="op">=</span><span class="va">gtfSource</span>, convertEnsemblUCSC<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Celegans_TSSs</span> </span>
<span><span class="co">#&gt; GRanges object with 19997 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source     type     score</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9246360-9246361      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [2]     chrV 11470663-11470664      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [3]     chrV 15817846-15817847      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [4]     chrV 20558370-20558371      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [5]     chrV     330725-330726      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...      ...       ...</span></span>
<span><span class="co">#&gt;   [19993] chrMtDNA         6505-6506      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19994] chrMtDNA         7826-7827      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19995] chrMtDNA         9648-9649      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19996] chrMtDNA       11355-11356      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19997] chrMtDNA       11690-11691      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;               phase        gene_id   gene_name gene_source   gene_biotype</span></span>
<span><span class="co">#&gt;           &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [2]      &lt;NA&gt; WBGene00000007       aat-6    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [3]      &lt;NA&gt; WBGene00000014       abf-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [4]      &lt;NA&gt; WBGene00000015       abf-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [5]      &lt;NA&gt; WBGene00000022       abt-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       ...       ...            ...         ...         ...            ...</span></span>
<span><span class="co">#&gt;   [19993]      &lt;NA&gt; WBGene00010963      nduo-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19994]      &lt;NA&gt; WBGene00010964       ctc-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19995]      &lt;NA&gt; WBGene00010965       ctc-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19996]      &lt;NA&gt; WBGene00010966      nduo-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19997]      &lt;NA&gt; WBGene00010967      nduo-5    WormBase protein_coding</span></span>
<span><span class="co">#&gt;           transcript_id transcript_source transcript_biotype exon_number</span></span>
<span><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...           ...               ...                ...         ...</span></span>
<span><span class="co">#&gt;   [19993]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;               exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [19993]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The output is GRanges object with TSSs coordinates, that can now be
passed to <em>calcFeatureDist</em> function and you get equivalent
results as if you used <em>calcFeatureDistRefTSS</em> function with your
genome of interest.</p>
</div>
<div class="section level5">
<h5 id="get-genome-partitions">4. Get genome partitions<a class="anchor" aria-label="anchor" href="#get-genome-partitions"></a>
</h5>
<p>For overlaps with genomic partitions: _ calcPartitions_, _
calcExpectedPartitions_, and _ calcCumulativePartitions_ functions, a
user needs to provide a GRangesList with coordinates for each partition.
We do that by first extracting gene models (coordinates for genes, exons
available 3’UTRs and 5’UTRs). Let’s start with extracting gene
models.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">features</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"exon"</span>,<span class="st">"three_prime_utr"</span>, <span class="st">"five_prime_utr"</span><span class="op">)</span></span>
<span><span class="va">Celegans_geneModels</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/getGeneModelsFromGTF.html" class="external-link">getGeneModelsFromGTF</a></span><span class="op">(</span>source<span class="op">=</span><span class="va">gtfSource</span>, features<span class="op">=</span><span class="va">features</span>, convertEnsemblUCSC<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">Celegans_geneModels</span></span>
<span><span class="co">#&gt; $gene</span></span>
<span><span class="co">#&gt; GRanges object with 19997 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source     type     score</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9244402-9246360      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [2]     chrV 11466842-11470663      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [3]     chrV 15817410-15817846      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [4]     chrV 20557876-20558370      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [5]     chrV     323777-330725      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...      ...       ...</span></span>
<span><span class="co">#&gt;   [19993] chrMtDNA         6506-7808      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19994] chrMtDNA         7827-9428      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19995] chrMtDNA        9649-10348      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19996] chrMtDNA       11356-11691      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19997] chrMtDNA       11691-13272      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;               phase        gene_id   gene_name gene_source   gene_biotype</span></span>
<span><span class="co">#&gt;           &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [2]      &lt;NA&gt; WBGene00000007       aat-6    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [3]      &lt;NA&gt; WBGene00000014       abf-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [4]      &lt;NA&gt; WBGene00000015       abf-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [5]      &lt;NA&gt; WBGene00000022       abt-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       ...       ...            ...         ...         ...            ...</span></span>
<span><span class="co">#&gt;   [19993]      &lt;NA&gt; WBGene00010963      nduo-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19994]      &lt;NA&gt; WBGene00010964       ctc-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19995]      &lt;NA&gt; WBGene00010965       ctc-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19996]      &lt;NA&gt; WBGene00010966      nduo-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19997]      &lt;NA&gt; WBGene00010967      nduo-5    WormBase protein_coding</span></span>
<span><span class="co">#&gt;           transcript_id transcript_source transcript_biotype exon_number</span></span>
<span><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...           ...               ...                ...         ...</span></span>
<span><span class="co">#&gt;   [19993]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;               exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [19993]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $exon</span></span>
<span><span class="co">#&gt; GRanges object with 144326 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames            ranges strand |   source     type     score</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;        [1]     chrV   9246080-9246360      - | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;        [2]     chrV   9245588-9246033      - | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;        [3]     chrV   9245443-9245539      - | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;        [4]     chrV   9244402-9245315      - | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;        [5]     chrV 11470583-11470663      - | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;        ...      ...               ...    ... .      ...      ...       ...</span></span>
<span><span class="co">#&gt;   [144322] chrMtDNA         6506-7808      + | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;   [144323] chrMtDNA         7827-9428      + | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;   [144324] chrMtDNA        9649-10348      + | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;   [144325] chrMtDNA       11356-11691      + | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;   [144326] chrMtDNA       11691-13272      + | WormBase     exon        NA</span></span>
<span><span class="co">#&gt;                phase        gene_id   gene_name gene_source   gene_biotype</span></span>
<span><span class="co">#&gt;            &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;        [1]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;        [2]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;        [3]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;        [4]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;        [5]      &lt;NA&gt; WBGene00000007       aat-6    WormBase protein_coding</span></span>
<span><span class="co">#&gt;        ...       ...            ...         ...         ...            ...</span></span>
<span><span class="co">#&gt;   [144322]      &lt;NA&gt; WBGene00010963      nduo-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [144323]      &lt;NA&gt; WBGene00010964       ctc-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [144324]      &lt;NA&gt; WBGene00010965       ctc-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [144325]      &lt;NA&gt; WBGene00010966      nduo-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [144326]      &lt;NA&gt; WBGene00010967      nduo-5    WormBase protein_coding</span></span>
<span><span class="co">#&gt;            transcript_id transcript_source transcript_biotype exon_number</span></span>
<span><span class="co">#&gt;              &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;        [1]     F07C3.7.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;        [2]     F07C3.7.1          WormBase     protein_coding           2</span></span>
<span><span class="co">#&gt;        [3]     F07C3.7.1          WormBase     protein_coding           3</span></span>
<span><span class="co">#&gt;        [4]     F07C3.7.1          WormBase     protein_coding           4</span></span>
<span><span class="co">#&gt;        [5]    T11F9.4a.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;        ...           ...               ...                ...         ...</span></span>
<span><span class="co">#&gt;   [144322]     MTCE.25.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;   [144323]     MTCE.26.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;   [144324]     MTCE.31.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;   [144325]     MTCE.34.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;   [144326]     MTCE.35.1          WormBase     protein_coding           1</span></span>
<span><span class="co">#&gt;                  exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;              &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;        [1]  F07C3.7.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;        [2]  F07C3.7.1.e2        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;        [3]  F07C3.7.1.e3        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;        [4]  F07C3.7.1.e4        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;        [5] T11F9.4a.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;        ...           ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [144322]  MTCE.25.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [144323]  MTCE.26.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [144324]  MTCE.31.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [144325]  MTCE.34.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [144326]  MTCE.35.1.e1        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $three_prime_utr</span></span>
<span><span class="co">#&gt; GRanges object with 23551 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source            type</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;        &lt;factor&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9244402-9244679      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [2]     chrV 11466842-11466969      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [3]     chrV 20557876-20557944      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [4]     chrV     323777-323909      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [5]     chrV 10380340-10380425      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...             ...</span></span>
<span><span class="co">#&gt;   [23547] chrMtDNA         7733-7808      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23548] chrMtDNA         9420-9428      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23549] chrMtDNA       10342-10348      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23550] chrMtDNA       11689-11691      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23551] chrMtDNA             13272      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;               score     phase        gene_id   gene_name gene_source</span></span>
<span><span class="co">#&gt;           &lt;numeric&gt; &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        NA      &lt;NA&gt; WBGene00000003       aat-2    WormBase</span></span>
<span><span class="co">#&gt;       [2]        NA      &lt;NA&gt; WBGene00000007       aat-6    WormBase</span></span>
<span><span class="co">#&gt;       [3]        NA      &lt;NA&gt; WBGene00000015       abf-4    WormBase</span></span>
<span><span class="co">#&gt;       [4]        NA      &lt;NA&gt; WBGene00000022       abt-4    WormBase</span></span>
<span><span class="co">#&gt;       [5]        NA      &lt;NA&gt; WBGene00000024       abu-1    WormBase</span></span>
<span><span class="co">#&gt;       ...       ...       ...            ...         ...         ...</span></span>
<span><span class="co">#&gt;   [23547]        NA      &lt;NA&gt; WBGene00010963      nduo-4    WormBase</span></span>
<span><span class="co">#&gt;   [23548]        NA      &lt;NA&gt; WBGene00010964       ctc-1    WormBase</span></span>
<span><span class="co">#&gt;   [23549]        NA      &lt;NA&gt; WBGene00010965       ctc-2    WormBase</span></span>
<span><span class="co">#&gt;   [23550]        NA      &lt;NA&gt; WBGene00010966      nduo-3    WormBase</span></span>
<span><span class="co">#&gt;   [23551]        NA      &lt;NA&gt; WBGene00010967      nduo-5    WormBase</span></span>
<span><span class="co">#&gt;             gene_biotype transcript_id transcript_source transcript_biotype</span></span>
<span><span class="co">#&gt;              &lt;character&gt;   &lt;character&gt;       &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1] protein_coding     F07C3.7.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [2] protein_coding    T11F9.4a.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [3] protein_coding   Y38H6C.22.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [4] protein_coding    Y39D8C.1.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [5] protein_coding       AC3.3.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       ...            ...           ...               ...                ...</span></span>
<span><span class="co">#&gt;   [23547] protein_coding     MTCE.25.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23548] protein_coding     MTCE.26.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23549] protein_coding     MTCE.31.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23550] protein_coding     MTCE.34.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23551] protein_coding     MTCE.35.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;           exon_number     exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [23547]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23548]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23549]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23550]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23551]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $five_prime_utr</span></span>
<span><span class="co">#&gt; GRanges object with 23688 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source           type     score</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;       &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9246353-9246360      - | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;       [2]     chrV 11470599-11470663      - | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;       [3]     chrV            330725      - | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;       [4]     chrV 10381786-10381826      - | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;       [5]     chrV   2274842-2274881      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...            ...       ...</span></span>
<span><span class="co">#&gt;   [23684]   chrIII   1987993-1988014      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;   [23685]   chrIII   2277142-2277278      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;   [23686]   chrIII   2280899-2280920      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;   [23687] chrMtDNA         3389-3417      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;   [23688] chrMtDNA         7827-7844      + | WormBase five_prime_utr        NA</span></span>
<span><span class="co">#&gt;               phase        gene_id   gene_name gene_source   gene_biotype</span></span>
<span><span class="co">#&gt;           &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [2]      &lt;NA&gt; WBGene00000007       aat-6    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [3]      &lt;NA&gt; WBGene00000022       abt-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [4]      &lt;NA&gt; WBGene00000024       abu-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [5]      &lt;NA&gt; WBGene00000027       abu-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       ...       ...            ...         ...         ...            ...</span></span>
<span><span class="co">#&gt;   [23684]      &lt;NA&gt; WBGene00306005    Y39A3B.8    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [23685]      &lt;NA&gt; WBGene00306012 Y54F10BM.21    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [23686]      &lt;NA&gt; WBGene00306013 Y54F10BM.22    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [23687]      &lt;NA&gt; WBGene00010961      nduo-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [23688]      &lt;NA&gt; WBGene00010964       ctc-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;           transcript_id transcript_source transcript_biotype exon_number</span></span>
<span><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]     F07C3.7.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]    T11F9.4a.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]    Y39D8C.1.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]       AC3.3.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]     Y5H2A.3.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...           ...               ...                ...         ...</span></span>
<span><span class="co">#&gt;   [23684]    Y39A3B.8.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23685] Y54F10BM.21.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23686] Y54F10BM.22.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23687]     MTCE.16.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23688]     MTCE.26.1          WormBase     protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;               exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [23684]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23685]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23686]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23687]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23688]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>We now need to extract additional elements: * core promoters (100 bp
upstream) * proximal promoters (2 kb upstream) * exons * introns * 5’UTR
(if provided) * 3’UTR (if provided)</p>
<p>This is all done automatically with following function:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">partitionList</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/genomePartitionList.html" class="external-link">genomePartitionList</a></span><span class="op">(</span>genesGR <span class="op">=</span> <span class="va">Celegans_geneModels</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>                                    exonsGR <span class="op">=</span> <span class="va">Celegans_geneModels</span><span class="op">$</span><span class="va">exon</span>, </span>
<span>                                    threeUTRGR <span class="op">=</span> <span class="va">Celegans_geneModels</span><span class="op">$</span><span class="va">three_prime_utr</span>,</span>
<span>                                    fiveUTRGR <span class="op">=</span> <span class="va">Celegans_geneModels</span><span class="op">$</span><span class="va">five_prime_utr</span><span class="op">)</span></span>
<span><span class="va">partitionList</span></span>
<span><span class="co">#&gt; $promoterCore</span></span>
<span><span class="co">#&gt; GRanges object with 19997 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source     type     score</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9246361-9246460      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [2]     chrV 11470664-11470763      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [3]     chrV 15817847-15817946      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [4]     chrV 20558371-20558470      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       [5]     chrV     330726-330825      - | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...      ...       ...</span></span>
<span><span class="co">#&gt;   [19993] chrMtDNA         6406-6505      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19994] chrMtDNA         7727-7826      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19995] chrMtDNA         9549-9648      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19996] chrMtDNA       11256-11355      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;   [19997] chrMtDNA       11591-11690      + | WormBase     gene        NA</span></span>
<span><span class="co">#&gt;               phase        gene_id   gene_name gene_source   gene_biotype</span></span>
<span><span class="co">#&gt;           &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]      &lt;NA&gt; WBGene00000003       aat-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [2]      &lt;NA&gt; WBGene00000007       aat-6    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [3]      &lt;NA&gt; WBGene00000014       abf-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [4]      &lt;NA&gt; WBGene00000015       abf-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       [5]      &lt;NA&gt; WBGene00000022       abt-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;       ...       ...            ...         ...         ...            ...</span></span>
<span><span class="co">#&gt;   [19993]      &lt;NA&gt; WBGene00010963      nduo-4    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19994]      &lt;NA&gt; WBGene00010964       ctc-1    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19995]      &lt;NA&gt; WBGene00010965       ctc-2    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19996]      &lt;NA&gt; WBGene00010966      nduo-3    WormBase protein_coding</span></span>
<span><span class="co">#&gt;   [19997]      &lt;NA&gt; WBGene00010967      nduo-5    WormBase protein_coding</span></span>
<span><span class="co">#&gt;           transcript_id transcript_source transcript_biotype exon_number</span></span>
<span><span class="co">#&gt;             &lt;character&gt;       &lt;character&gt;        &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...           ...               ...                ...         ...</span></span>
<span><span class="co">#&gt;   [19993]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]          &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;               exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [19993]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19994]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19995]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19996]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [19997]        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $promoterProx</span></span>
<span><span class="co">#&gt; GRanges object with 19907 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames      ranges strand</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrI  9495-11394      +</span></span>
<span><span class="co">#&gt;       [2]     chrI 41733-43632      +</span></span>
<span><span class="co">#&gt;       [3]     chrI 45461-47360      +</span></span>
<span><span class="co">#&gt;       [4]     chrI 47921-49820      +</span></span>
<span><span class="co">#&gt;       [5]     chrI 69425-71324      +</span></span>
<span><span class="co">#&gt;       ...      ...         ...    ...</span></span>
<span><span class="co">#&gt;   [19903] chrMtDNA   5678-6405      +</span></span>
<span><span class="co">#&gt;   [19904] chrMtDNA   6506-7726      +</span></span>
<span><span class="co">#&gt;   [19905] chrMtDNA   7827-9548      +</span></span>
<span><span class="co">#&gt;   [19906] chrMtDNA  9649-11255      +</span></span>
<span><span class="co">#&gt;   [19907] chrMtDNA 11356-11590      +</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $threeUTR</span></span>
<span><span class="co">#&gt; GRanges object with 23551 ranges and 16 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand |   source            type</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;        &lt;factor&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrV   9244402-9244679      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [2]     chrV 11466842-11466969      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [3]     chrV 20557876-20557944      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [4]     chrV     323777-323909      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       [5]     chrV 10380340-10380425      - | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ... .      ...             ...</span></span>
<span><span class="co">#&gt;   [23547] chrMtDNA         7733-7808      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23548] chrMtDNA         9420-9428      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23549] chrMtDNA       10342-10348      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23550] chrMtDNA       11689-11691      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;   [23551] chrMtDNA             13272      + | WormBase three_prime_utr</span></span>
<span><span class="co">#&gt;               score     phase        gene_id   gene_name gene_source</span></span>
<span><span class="co">#&gt;           &lt;numeric&gt; &lt;integer&gt;    &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        NA      &lt;NA&gt; WBGene00000003       aat-2    WormBase</span></span>
<span><span class="co">#&gt;       [2]        NA      &lt;NA&gt; WBGene00000007       aat-6    WormBase</span></span>
<span><span class="co">#&gt;       [3]        NA      &lt;NA&gt; WBGene00000015       abf-4    WormBase</span></span>
<span><span class="co">#&gt;       [4]        NA      &lt;NA&gt; WBGene00000022       abt-4    WormBase</span></span>
<span><span class="co">#&gt;       [5]        NA      &lt;NA&gt; WBGene00000024       abu-1    WormBase</span></span>
<span><span class="co">#&gt;       ...       ...       ...            ...         ...         ...</span></span>
<span><span class="co">#&gt;   [23547]        NA      &lt;NA&gt; WBGene00010963      nduo-4    WormBase</span></span>
<span><span class="co">#&gt;   [23548]        NA      &lt;NA&gt; WBGene00010964       ctc-1    WormBase</span></span>
<span><span class="co">#&gt;   [23549]        NA      &lt;NA&gt; WBGene00010965       ctc-2    WormBase</span></span>
<span><span class="co">#&gt;   [23550]        NA      &lt;NA&gt; WBGene00010966      nduo-3    WormBase</span></span>
<span><span class="co">#&gt;   [23551]        NA      &lt;NA&gt; WBGene00010967      nduo-5    WormBase</span></span>
<span><span class="co">#&gt;             gene_biotype transcript_id transcript_source transcript_biotype</span></span>
<span><span class="co">#&gt;              &lt;character&gt;   &lt;character&gt;       &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1] protein_coding     F07C3.7.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [2] protein_coding    T11F9.4a.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [3] protein_coding   Y38H6C.22.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [4] protein_coding    Y39D8C.1.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       [5] protein_coding       AC3.3.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;       ...            ...           ...               ...                ...</span></span>
<span><span class="co">#&gt;   [23547] protein_coding     MTCE.25.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23548] protein_coding     MTCE.26.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23549] protein_coding     MTCE.31.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23550] protein_coding     MTCE.34.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;   [23551] protein_coding     MTCE.35.1          WormBase     protein_coding</span></span>
<span><span class="co">#&gt;           exon_number     exon_id  protein_id transcript_name         tag</span></span>
<span><span class="co">#&gt;           &lt;character&gt; &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [2]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [3]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [4]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       [5]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;       ...         ...         ...         ...             ...         ...</span></span>
<span><span class="co">#&gt;   [23547]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23548]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23549]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23550]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   [23551]        &lt;NA&gt;        &lt;NA&gt;        &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $fiveUTR</span></span>
<span><span class="co">#&gt; GRanges object with 19396 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames            ranges strand</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;       [1]     chrI       11495-11561      +</span></span>
<span><span class="co">#&gt;       [2]     chrI       11618-11640      +</span></span>
<span><span class="co">#&gt;       [3]     chrI       47461-47471      +</span></span>
<span><span class="co">#&gt;       [4]     chrI       71845-71932      +</span></span>
<span><span class="co">#&gt;       [5]     chrI       72511-72590      +</span></span>
<span><span class="co">#&gt;       ...      ...               ...    ...</span></span>
<span><span class="co">#&gt;   [19392]     chrX 17682295-17682305      -</span></span>
<span><span class="co">#&gt;   [19393]     chrX 17684756-17684811      -</span></span>
<span><span class="co">#&gt;   [19394]     chrX 17711675-17711698      -</span></span>
<span><span class="co">#&gt;   [19395] chrMtDNA         3389-3417      +</span></span>
<span><span class="co">#&gt;   [19396] chrMtDNA         7827-7844      +</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $exon</span></span>
<span><span class="co">#&gt; GRanges object with 120584 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames      ranges strand</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;        [1]     chrI 11641-11689      +</span></span>
<span><span class="co">#&gt;        [2]     chrI 14951-15160      +</span></span>
<span><span class="co">#&gt;        [3]     chrI 16473-16585      +</span></span>
<span><span class="co">#&gt;        [4]     chrI 43733-43961      +</span></span>
<span><span class="co">#&gt;        [5]     chrI 44030-44234      +</span></span>
<span><span class="co">#&gt;        ...      ...         ...    ...</span></span>
<span><span class="co">#&gt;   [120580] chrMtDNA   6506-7732      +</span></span>
<span><span class="co">#&gt;   [120581] chrMtDNA   7845-9419      +</span></span>
<span><span class="co">#&gt;   [120582] chrMtDNA  9649-10341      +</span></span>
<span><span class="co">#&gt;   [120583] chrMtDNA 11356-11688      +</span></span>
<span><span class="co">#&gt;   [120584] chrMtDNA 11692-13271      +</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $intron</span></span>
<span><span class="co">#&gt; GRanges object with 104757 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames            ranges strand</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;        [1]     chrI       11562-11617      +</span></span>
<span><span class="co">#&gt;        [2]     chrI       11690-14950      +</span></span>
<span><span class="co">#&gt;        [3]     chrI       15161-16472      +</span></span>
<span><span class="co">#&gt;        [4]     chrI       43962-44029      +</span></span>
<span><span class="co">#&gt;        [5]     chrI       44235-44280      +</span></span>
<span><span class="co">#&gt;        ...      ...               ...    ...</span></span>
<span><span class="co">#&gt;   [104753]     chrX 17682306-17684755      -</span></span>
<span><span class="co">#&gt;   [104754]     chrX 17710702-17710749      -</span></span>
<span><span class="co">#&gt;   [104755]     chrX 17710904-17711062      -</span></span>
<span><span class="co">#&gt;   [104756]     chrX 17711269-17711380      -</span></span>
<span><span class="co">#&gt;   [104757]     chrX 17711486-17711596      -</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>If you want additional features included in the partition list,
simply attach them as additional <em>GRanges</em> to the existing
list.</p>
<p>And that’s it! We now have all genome annotations required to run all
of the <em>GenomicDistributions</em> functions! Let’s try them!</p>
</div>
<div class="section level5">
<h5 id="load-c--elegans-region-sets">Load <em>C. elegans</em> region sets<a class="anchor" aria-label="anchor" href="#load-c--elegans-region-sets"></a>
</h5>
<p>First load data into GRangesList object just like before.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># provide full path to genomic regions of interest (BED files)</span></span>
<span><span class="va">elt3_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ce11_elt3.bed.gz"</span>, </span>
<span>                                 package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span><span class="va">ztf11_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ce11_ztf11.bed.gz"</span>, </span>
<span>                                  package<span class="op">=</span><span class="st">"GenomicDistributionsBioC2022"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># import bed files as GenomicRange objects</span></span>
<span><span class="va">elt3</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">elt3_file</span><span class="op">)</span></span>
<span><span class="va">ztf11</span> <span class="op">=</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="va">ztf11_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove unknown chromosomes:</span></span>
<span><span class="va">elt3</span> <span class="op">=</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">elt3</span>,pruning.mode<span class="op">=</span><span class="st">"coarse"</span><span class="op">)</span></span>
<span><span class="va">ztf11</span> <span class="op">=</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">ztf11</span>,pruning.mode<span class="op">=</span><span class="st">"coarse"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">GRList_CE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></span><span class="op">(</span>elt3<span class="op">=</span><span class="va">elt3</span>,</span>
<span>                        ztf11<span class="op">=</span><span class="va">ztf11</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="distribution-over-chromosomes-calcchrombins-plotchrombins">Distribution over chromosomes: <em>calcChromBins()</em> +
<em>plotChromBins()</em><a class="anchor" aria-label="anchor" href="#distribution-over-chromosomes-calcchrombins-plotchrombins"></a>
</h5>
<p>Let’s start by plotting the distribution over chromosomes. Notice
that the used <em>calc</em> function is missing the <em>Ref</em> suffix,
since we are providing our own annotations in form of the previously
created bins over genome.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDistribution_CE</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcChromBins.html" class="external-link">calcChromBins</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList_CE</span>, bins<span class="op">=</span><span class="va">Celegans_bins</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotChromBins.html" class="external-link">plotChromBins</a></span><span class="op">(</span><span class="va">chromDistribution_CE</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-plots-CE-1.png" width="288"></p>
</div>
<div class="section level5">
<h5 id="distances-from-features-calcfeaturedist-plotfeaturedist">Distances from features: <em>calcFeatureDist()</em> +
<em>plotFeatureDist()</em><a class="anchor" aria-label="anchor" href="#distances-from-features-calcfeaturedist-plotfeaturedist"></a>
</h5>
<p><strong>Distances from TSSs</strong></p>
<p>Now that we have the coordinates of TSSs in <em>C. elegans</em>,
let’s have a look at the distances of our region sets to TSSs.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TSSdist_CE</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcFeatureDist.html" class="external-link">calcFeatureDist</a></span><span class="op">(</span><span class="va">GRList_CE</span>, features<span class="op">=</span><span class="va">Celegans_TSSs</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotFeatureDist.html" class="external-link">plotFeatureDist</a></span><span class="op">(</span><span class="va">TSSdist_CE</span>, featureName<span class="op">=</span><span class="st">"TSS"</span>, size <span class="op">=</span> <span class="fl">5000</span>, nbins <span class="op">=</span> <span class="fl">50</span>, infBins <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/TSSdistCE-1.png" width="576"></p>
<p><strong><em>Exercise 2</em></strong></p>
<p>The <em>calcFeatureDist</em> function is not designed only for
distances from TSSs. Can you calculate and plot the distances of elt3
regions from ztf11? You can then test the different <em>size</em> /
<em>nbins</em> / <em>infBins</em> settings in the
<em>plotFeatureDist</em> function to get an optimal view. And don’t
foget to change the <em>featureName</em> in the <em>plotFeatureDist</em>
function to “ztf3”.</p>
</div>
<div class="section level5">
<h5 id="distribution-over-genomic-partitions-calcpartitions-plotpartitions">Distribution over genomic partitions: <em>calcPartitions()</em> +
<em>plotPartitions()</em><a class="anchor" aria-label="anchor" href="#distribution-over-genomic-partitions-calcpartitions-plotpartitions"></a>
</h5>
<p>Next we can use the create _ partitionList_ to get overlaps with
genomic partitions.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gp</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcPartitions.html" class="external-link">calcPartitions</a></span><span class="op">(</span><span class="va">GRList_CE</span>, partitionList <span class="op">=</span> <span class="va">partitionList</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotPartitions.html" class="external-link">plotPartitions</a></span><span class="op">(</span><span class="va">gp</span><span class="op">)</span> </span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/partitionsCE-1.png" width="384"></p>
<p><strong><em>Exercise 3</em></strong></p>
<p>With generated <em>partitionList</em>, can you plot cumulative
partition distribution for the <em>C. elegans</em> regions? Use
<em>calcCumulativePartitions</em> function to do the calculations and
<em>plotCumulativePartitions</em> function to plot the results.</p>
</div>
</div>
<div class="section level3">
<h3 id="k--customize-plotting">K. Customize plotting<a class="anchor" aria-label="anchor" href="#k--customize-plotting"></a>
</h3>
<p>Would you like to change colors, title, axis labels, etc. in your
plots? No problem. Many of <em>GenomicDistributions</em> plotting
functions offer various options. But that’s not all. The output from the
plotting functions are editable <em>ggplot</em> objects. Let’s try to
change few things.</p>
<p>Let’s plot again the results showing distances between nearest
neighbors.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">neighbors</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcNeighborDist.html" class="external-link">calcNeighborDist</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotNeighborDist.html" class="external-link">plotNeighborDist</a></span><span class="op">(</span><span class="va">neighbors</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/neighbors2-1.png" width="576"></p>
<p>Now, let’s change our color scheme, maybe title and let’s put the
legend on the right.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotNeighborDist.html" class="external-link">plotNeighborDist</a></span><span class="op">(</span><span class="va">neighbors</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#A0D8F1"</span>, <span class="st">"#E9AF32"</span>,<span class="st">"#BF381A"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#A0D8F1"</span>, <span class="st">"#E9AF32"</span>,<span class="st">"#BF381A"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"This is my new title"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/neighborsEdit-1.png" width="576"></p>
<p>Changing color options in a plot showing summary for one region set
is maybe a bit tricky, but definitely possible. Let’s try it! Let’s
start by plotting the original.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">singleNeighbors</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcNearestNeighbors.html" class="external-link">calcNearestNeighbors</a></span><span class="op">(</span><span class="va">FGF2_iPSC</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotNeighborDist.html" class="external-link">plotNeighborDist</a></span><span class="op">(</span><span class="va">singleNeighbors</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/neighborsSingle-1.png" width="576"></p>
<p>This plot is not very appealing. Lets’ make the colors nicer. And
maybe change x-axis label.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">singleNeighbors</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcNearestNeighbors.html" class="external-link">calcNearestNeighbors</a></span><span class="op">(</span><span class="va">FGF2_iPSC</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotNeighborDist.html" class="external-link">plotNeighborDist</a></span><span class="op">(</span><span class="va">singleNeighbors</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">=</span> <span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"distnace [bp]"</span><span class="op">)</span></span>
<span><span class="va">p</span><span class="op">$</span><span class="va">layers</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">aes_params</span><span class="op">$</span><span class="va">fill</span> <span class="op">=</span>  <span class="st">"#A0D8F1"</span></span>
<span><span class="va">p</span><span class="op">$</span><span class="va">layers</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">aes_params</span><span class="op">$</span><span class="va">colour</span> <span class="op">=</span>  <span class="st">"#E9AF32"</span></span>
<span><span class="va">p</span><span class="op">$</span><span class="va">layers</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">aes_params</span><span class="op">$</span><span class="va">alpha</span> <span class="op">=</span>  <span class="fl">0.5</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/neighborsSingleEdit-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="additional-examples">Additional examples<a class="anchor" aria-label="anchor" href="#additional-examples"></a>
</h3>
<div class="section level4">
<h4 id="change-the-resolution-of-distribution-over-chromosomes">Change the resolution of distribution over chromosomes<a class="anchor" aria-label="anchor" href="#change-the-resolution-of-distribution-over-chromosomes"></a>
</h4>
<p>How can I change the resolution of the distribution of regions over
chromosome? Simply by changing <em>binCount</em> argument in the
<em>calcChromBins</em> / <em>calcChromBinsRef</em> functions.</p>
<p>Let’s first set the number of bins (within the whole genome) to
10,000.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDistributionMulti</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcChromBinsRef.html" class="external-link">calcChromBinsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span>, binCount<span class="op">=</span><span class="fl">100000</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotChromBins.html" class="external-link">plotChromBins</a></span><span class="op">(</span><span class="va">chromDistributionMulti</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-plots-multi-hgh-resolution-1.png" width="480"></p>
<p>And now let’s have a look, how does the resolution change by setting
the number of bins to 1,000.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromDistributionMulti</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcChromBinsRef.html" class="external-link">calcChromBinsRef</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span>, binCount<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotChromBins.html" class="external-link">plotChromBins</a></span><span class="op">(</span><span class="va">chromDistributionMulti</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/chrom-plots-multi-lw-resolution-1.png" width="480"></p>
</div>
<div class="section level4">
<h4 id="plot-distances-from-tsss-in-form-of-heatmap-instead-of-histogram">Plot distances from TSSs in form of heatmap instead of
histogram<a class="anchor" aria-label="anchor" href="#plot-distances-from-tsss-in-form-of-heatmap-instead-of-histogram"></a>
</h4>
<p>Instead of plotting the distances from TSSs in form of histogram,
let’s convert the bar heights into colors and plot the results in form
of heatmap (set up argument <code>tile=TRUE</code>). In this example, we
will also set up the order of the samples based on the value in the plot
center (<code>labelOrder="center"</code>).</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TSSdist</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcFeatureDistRefTSS.html" class="external-link">calcFeatureDistRefTSS</a></span><span class="op">(</span>query<span class="op">=</span><span class="va">GRList</span>, refAssembly<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotFeatureDist.html" class="external-link">plotFeatureDist</a></span><span class="op">(</span><span class="va">TSSdist</span>, featureName<span class="op">=</span><span class="st">"TSS"</span>, size<span class="op">=</span><span class="fl">5000</span>, labelOrder<span class="op">=</span><span class="st">"center"</span>, nbins<span class="op">=</span><span class="fl">50</span>, tile<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/TSS-heatmap-1.png" width="480"></p>
</div>
<div class="section level4">
<h4 id="custom-features">Custom features<a class="anchor" aria-label="anchor" href="#custom-features"></a>
</h4>
</div>
</div>
<div class="section level3">
<h3 id="exercise-solutions">Exercise solutions<a class="anchor" aria-label="anchor" href="#exercise-solutions"></a>
</h3>
<p><strong><em>Exercise 1</em></strong></p>
<p>Using the previously calculated _ TSSdist_ object, plot histogram of
distances to TSSs within 5 kb from TSSs (argument <em>size</em>), where
regions further than +-5 kb will be accumulated and plotted into
“infinite bins” (argument <em>infBins</em>).</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># TSSdist = calcFeatureDistRefTSS(query=GRList, refAssembly="hg38")</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotFeatureDist.html" class="external-link">plotFeatureDist</a></span><span class="op">(</span><span class="va">TSSdist</span>, featureName<span class="op">=</span><span class="st">"TSS"</span>, size<span class="op">=</span><span class="fl">5000</span>, infBins<span class="op">=</span><span class="cn">T</span><span class="op">)</span> </span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/Exiercise1-1.png" width="480"></p>
<p><strong><em>Exercise 2</em></strong></p>
<p>The <em>calcFeatureDist</em> function is not designed only for
distances from TSSs. Can you calculate and plot the distances of elt3
regions from ztf11? You can then test the different <em>size</em> /
<em>nbins</em> / <em>infBins</em> settings in the
<em>plotFeatureDist</em> function to get an optimal view. And don’t
foget to change the <em>featureName</em> in the <em>plotFeatureDist</em>
function to “ztf3”.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist_ztf11</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcFeatureDist.html" class="external-link">calcFeatureDist</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">elt3</span>, features <span class="op">=</span> <span class="va">ztf11</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotFeatureDist.html" class="external-link">plotFeatureDist</a></span><span class="op">(</span><span class="va">dist_ztf11</span>, featureName<span class="op">=</span><span class="st">"ztf11"</span>, size <span class="op">=</span> <span class="fl">1000000</span>, nbins <span class="op">=</span> <span class="fl">50</span>, infBins <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/Exiercise2-1.png" width="288"></p>
<p><strong><em>Exercise 3</em></strong></p>
<p>With generated <em>partitionList</em>, can you plot cumulative
partition distribution for the <em>C. elegans</em> regions? Use
<em>calcCumulativePartitions</em> function to do the calculations and
<em>plotCumulativePartitions</em> function to plot the results.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cp</span> <span class="op">=</span> <span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/calcCumulativePartitions.html" class="external-link">calcCumulativePartitions</a></span><span class="op">(</span><span class="va">GRList_CE</span>, partitionList <span class="op">=</span> <span class="va">partitionList</span><span class="op">)</span></span>
<span><span class="fu"><a href="http://code.databio.org/GenomicDistributions/reference/plotCumulativePartitions.html" class="external-link">plotCumulativePartitions</a></span><span class="op">(</span><span class="va">cp</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomicDistributions_vignette_files/figure-html/Exiercise3-1.png" width="576"></p>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:kristynakupkova@gmail.com" class="email">kristynakupkova@gmail.com</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Center for Public Health Genomics / Department of
Biochemistry and Molecular Genetics, University of Virginia<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Center for Public Health Genomics / Department of
Biochemistry and Molecular Genetics, University of Virginia<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Center for Public Health Genomics / Department of
Biochemistry and Molecular Genetics, University of Virginia<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>Center for Public Health Genomics, University of
Virginia<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p>Center for Public Health Genomics, University of
Virginia<a href="#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p>Center for Public Health Genomics / Department of
Biomedical Engineering, University of Virginia<a href="#fnref7" class="footnote-back">↩︎</a></p></li>
<li id="fn8"><p>Center for Public Health Genomics / Department of
Biomedical Engineering, University of Virginia<a href="#fnref8" class="footnote-back">↩︎</a></p></li>
<li id="fn9"><p>Center for Public Health Genomics / Department of
Biochemistry and Molecular Genetics, University of Virginia<a href="#fnref9" class="footnote-back">↩︎</a></p></li>
<li id="fn10"><p>Center for Public Health Genomics / Department of
Biomedical Engineering, University of Virginia<a href="#fnref10" class="footnote-back">↩︎</a></p></li>
<li id="fn11"><p>Center for Public Health Genomics / Department of
Public Health Sciences / Department of Biomedical Engineering /
Department of Biochemistry and Molecular Genetics, University of
Virginia<a href="#fnref11" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kristyna Kupkova, Jose Verdezoto, Tessa Danehy, John Lawson, Jose Verdezoto, Michal Stolarczyk, Jason Smith, Bingjie Xue, Sophia Rogers, John Stubbs, Nathan C. Sheffield.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
